lib/ansible/modules/cloud/amazon/rds_tag.py

```diff
@@ -0,0 +1,270 @@
+                             msg="Couldn't remove tag to %s instance"
+                             % configured_tags['arn'])
+
+    response['message'] = 'Remove %s tag to %s instance' % (
+        tags.keys(), configured_tags['arn'])
+
+    return response
+
+
+def list_rds_arn(module, client, instance_name):
+    if instance_name is None:
+        try:
+            response_instances = client.describe_db_instances()
+        except (botocore.exceptions.ClientError,
+                botocore.exceptions.BotoCoreError) as e:
+            module.fail_json_aws(e, msg="Couldn't find RDS instance")
+    else:
+        try:
+            response_instances = client.describe_db_instances(
+                DBInstanceIdentifier=instance_name)
+        except (botocore.exceptions.ClientError,
+                botocore.exceptions.BotoCoreError) as e:
+            module.fail_json_aws(e, msg="Couldn't find %s instance"
+                                 % instance_name)
+
+    list_arn = [{'arn': instance['DBInstanceArn'],
+                 'identifier': instance['DBInstanceIdentifier']}
+                for instance in response_instances['DBInstances']]
+
+    return list_arn
+
+
+def list_rds_tags(module, client, list_arn, instance_name):
+    list_tags = []
+    for map_arn in list_arn:
+        try:
+            response_tags = client.list_tags_for_resource(
+                ResourceName=map_arn['arn'])
+        except (botocore.exceptions.ClientError,
+                botocore.exceptions.BotoCoreError) as e:
+            module.fail_json_aws(e, msg="Couldn't get %s instance's tag"
+                                 % instance_name)
+        tags = {}
+        for datum in response_tags['TagList']:
+            tags[datum['Key']] = datum['Value']
+
+        list_tags.append({'identifier': map_arn['identifier'],
+                          'tags': tags,
+                          'arn': map_arn['arn']})
+    return list_tags
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            instance_name=dict(type='str'),
+            tags=dict(type='dict'),
+            state=dict(default='present',
+                       type='str',
+                       choices=['present', 'absent', 'list'])
+        )
+    )
+
+    module = AnsibleAWSModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True)
+
+    region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module,
+                                                                  boto3=True)
+
+    client = boto3_conn(module, conn_type='client', resource='rds',
+                        region=region, **aws_connect_kwargs)
+
+    instance_name = module.params.get('instance_name')
+    tags = module.params.get('tags')
+    state = module.params.get('state')
+
+    list_arn = list_rds_arn(module, client, instance_name)
+    list_tags = list_rds_tags(module, client, list_arn, instance_name)
+
+    if state == 'list':
```

Comment:
This should go in a facts module instead.

Reply:
I fully agree with the opinion that the function of list should be done in the facts module.

Reply:
I fully agree with the opinion that the functions of the list should be done in fact modules.